### DISCUSSION TITLE
native chromecast support

### DISCUSSION THREAD

**Discussion by dietrichmd** [advplyr/audiobookshelf#367] [Ideas]
Title pretty much says it all.  Would be nice for native chromecast support on both the app and webapp.

**Comment by Jdiesel87** [comment]
I had made a FR over on the app project but it would probably be better suited here on the server side

https://github.com/advplyr/audiobookshelf-app/issues/11

**Reply by advplyr** [reply]
I got this partially implemented on the android app a few months ago but put it on hold because it was taking so much time. It is surprisingly difficult to support.. but I'm sure it'll get figured out.

**Reply by advplyr** [reply]
Now that I setup a chromecast receiver implementing the mobile apps shouldn't be as difficult. I didn't have a good understanding of how it worked initially.

**Comment by advplyr** [comment]
`v1.6.70` is released with chromecast support in experimental.

Enable experimental features will show the Enable Chromecast setting. After turning that on refresh the page. Make sure you are on https.
![image](https://user-images.githubusercontent.com/67830747/155335263-f1f98caf-cafd-401d-b694-8326acaf85dd.png)
![image](https://user-images.githubusercontent.com/67830747/155335423-c02c91a4-8cbc-4414-95ee-339c112c6b17.png)

When no audiobooks are being played the idle chromecast screen shows a rotating slideshow of your book covers. It turned out that in production the images load much slower, so it doesnt look very good on first pass.

**Reply by advplyr** [reply]
I'm also not sure if chromecast support playback speed changes. The settings didn't work and it looks like it may have been only recently added to the iOS API.

**Reply by dietrichmd** [reply]
Well, from Chrome in WIn10 to a google speaker, it works like a charm...at least basic playback.  Speed changes appear to work as well.

**Reply by advplyr** [reply]
That's great, thanks for testing that so quickly. What do you mean speed changes are working? When you set to 1.5x playback speed and start an audiobook it plays at 1.5x speed?

**Reply by Jdiesel87** [reply]
I don't know if it is a side effect of accessing my server from a remote network with no compatible devices but the icon is not visible. I can see when the space between the icons increases to make room for the chromecast icon but that's as far as it goes.

![Capture](https://user-images.githubusercontent.com/14824476/155338878-5d4fdc54-d29c-426b-b42f-cbaf6a69d062.PNG)

I will have to test it when I am back on my home network.

**Reply by advplyr** [reply]
Hmm, that's strange that there is a space there. Usually when there are no networks there is no space.
Are you accessing the web server over https?

**Reply by Jdiesel87** [reply]
Yes I am accessing over https. After a slight delay the space expands and my cursor switches to indicate there is a object there but clicking does nothing. At first I though it was just an asset failing to load so I trying doing a hard flush on my browser cache.

I just tried VPNing into my local network from my phone and I see the same space on mobile Chrome as well. Maybe it is a reverse proxy config issue on my end...

**Reply by advplyr** [reply]
Can you check the developer console to see if there are any errors there? Ctrl+Shift+i

**Reply by dietrichmd** [reply]
When you set to 1.5x playback speed and start an audiobook it plays at 1.5x speed?

Correct

**Reply by Jdiesel87** [reply]
> 
![Capture](https://user-images.githubusercontent.com/14824476/155341321-a6f35021-bfc9-4189-abd4-305e72beacad.PNG)

**Reply by advplyr** [reply]
Oh interesting, because it wasn't using the set playback speed on my device.
@Jdiesel87 have you tried choosing cast from the browser dropdown menu? if you open the browser to audiobookshelf then 
![image](https://user-images.githubusercontent.com/67830747/155346039-726158bf-d885-40f9-b17b-e9627ab3628e.png)

**Reply by Jdiesel87** [reply]
I will wait until I am home to properly troubleshoot, if needed at all. It very well could be a networking issue on my end too.

**Reply by Jdiesel87** [reply]
All is good on my home network. Thanks for implementing the feature! I will be testing it more over the next few days.

**Comment by advplyr** [comment]
Chromecast support is no longer experimental in [v1.7.0](https://github.com/advplyr/audiobookshelf/releases/tag/v1.7.0)

It seemed stable enough.

**Reply by kbipinkumar** [reply]
Any plans or ETA for chromecast support in iOS app?

**Reply by advplyr** [reply]
I haven't looked into it yet but yeah it'll probably be added. Car Play is more requested so that would be priority. Not that I always follow a set order.

**Reply by fivestones** [reply]
I'd be happy to have iOS chromecast support too whenever you get a chance.
A suggestion is that maybe it should be documented somewhere (maybe in the settings where you turn on/off chromecast support?) where you can expect casting to work (like, "works over https on android/chrome browser" or something). I thought it would work on iOS and tried a bunch of things before finding this and realizing it probably wasn't implemented there yet.
Thanks for all you do!
